### PR TITLE
AB#69 Transaction searching

### PR DESCRIPTION
**Pull request by Kesuera**
Added the transaction API with the ability to search for transactions based on given criteria alongside with pagination.

### CODE REVIEW

File: server/services/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useApi } from '../utils/api';
+import { TransactionDTO, TransactionSearchRequestDTO, TransactionSearchResponseDTO } from '~/types/dtos';
+
+export const useTransactionService = (token: string) => {
+  const getTransaction = async (id: string) => {
+    const data = await useApi<TransactionDTO>(`/transaction/${id}`, token);
+    return data;
+  };
+
+  const searchTransactions = async (query: TransactionSearchRequestDTO) => {
```
Comment (bran0h) [inline server/services/transaction.ts:9]: This could be one liner

File: server/services/transaction.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useApi } from '../utils/api';
+import { TransactionDTO, TransactionSearchRequestDTO, TransactionSearchResponseDTO } from '~/types/dtos';
+
+export const useTransactionService = (token: string) => {
+  const getTransaction = async (id: string) => {
```
Comment (bran0h) [inline server/services/transaction.ts:5]: Same

File: server/config/api.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 export const BASE_URL = 'https://localhost:7210';
```
Comment (bran0h) [inline server/config/api.ts]: Load this from env, this is enviroment specific URL

File: server/api/transaction/search.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useTransactionService } from '~/server/services/transaction';
+import { getToken } from '#auth';
+import { TransactionSearchRequestDTO } from '~/types/dtos';
+
+export default defineEventHandler(async (event) => {
+  const token = await getToken({ event });
```
Comment (bran0h) [inline server/api/transaction/search.ts:6]: Please examine if we can move this check to some middleware

File: server/api/transaction/[id].ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ export default defineEventHandler(async (event) => {
       statusMessage: 'Unauthorized',
     });
 
-  const { getForecasts } = useWeatherService(token.googleAccessToken);
-  return await getForecasts();
+  const { getTransaction } = useTransactionService(token.googleAccessToken);
+  const id = event.context.params!.id;
```
Comment (bran0h) [inline server/api/transaction/[id].ts]: There is some method like `getParams(event)` to do this safe

File: components/project/sections/TransactionsSection.vue [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        :loading="transactionsLoading"
+        :pagination="pagination"
+      />
+      <div v-else>Error: {{ error }}</div>
+    </n-space>
+  </n-card>
+</template>
+
+<script setup lang="ts">
+import { NDataTable, NCard, NButton, NSpace, NInput } from 'naive-ui';
```
Comment (bran0h) [inline components/project/sections/TransactionsSection.vue:28]: You dont have to do this in this task, but we should switch to nuxt module naiveui. There is much less config required and all the naive components are autoinjected. I tested it and it works like charm

**Review by bran0h — CHANGES_REQUESTED** [review]

File: server/services/transaction.ts (hunk 1 above)
Comment (Kesuera) [reply server/services/transaction.ts:9]: fixed

File: server/services/transaction.ts (hunk 2 above)
Comment (Kesuera) [reply server/services/transaction.ts:5]: fixed

File: server/config/api.ts (hunk 3 above)
Comment (Kesuera) [reply server/config/api.ts]: Moved both the BASE_URL and GOOGLE_REFRESH_TOKEN_URL to env

File: server/api/transaction/[id].ts (hunk 5 above)
Comment (Kesuera) [reply server/api/transaction/[id].ts]: getRouterParam(event, "id") exists, but you still have to retype that as string for typescript to be happy, I did that

File: server/services/transaction.ts (hunk 2 above)
Comment (bran0h) [reply server/services/transaction.ts:5]: Even shorter, remove the body

**Review by bran0h — CHANGES_REQUESTED** [review]
Shorter

**Review by bran0h — APPROVED** [review]

**Pull request merged**